src/java.base/share/classes/java/util/zip/ZipInputStream.java

```diff
@@ -38,9 +38,36 @@
 import static java.util.zip.ZipUtils.*;
 
 /**
- * This class implements an input stream filter for reading files in the
- * ZIP file format. Includes support for both compressed and uncompressed
- * entries.
+ * An input stream for reading compressed and uncompressed
+ * {@linkplain ZipEntry ZIP file entries} from a stream of bytes in the ZIP file
+ * format.
+ *
+ * <H2>Reading Zip File Entries</H2>
+ *
+ * The {@link #getNextEntry()} method is used to read the next ZIP file entry
+ * (Local file (LOC) header record in the ZIP format) and position the stream at
+ * the entry's file data. The file data may read using one of
+ * {@code ZipInputStream} read methods such
+ * as {@link #read(byte[], int, int) read} or {@link #readAllBytes() readAllBytes()}.
+ * For example:
+ * <pre>
+ *     {@code
+ *      try (FileInputStream fis = new FileInputStream(jar.toFile());
+ *              ZipInputStream zis = new ZipInputStream(fis)) {
```

Comment:
The indentation seems weird in this code sample.

Reply:
Reformatted